src/bootstrap/bootstrap.py

```diff
@@ -12,9 +12,25 @@
 import sys
 import tarfile
 import tempfile
+import sqlite3
 
 from time import time
 
+def acquire_lock():
+    try:
+        con = sqlite3.Connection("xlock.db", timeout=0)
```

Comment:
Is this creating the database in the working directory? I think we probably want it in build/ -- build.build_dir probably?

Reply:
I changed it to creating it in `build.rust_dir`: The lock locks a lot more than just the stuff in `build` (submodules, etc), and the folder may not exist at the time of lock creation.